pkg/p2p/libp2p/internal/handshake/handshake_test.go

```diff
@@ -241,8 +274,49 @@ func TestHandshake(t *testing.T) {
 		}
 	})
 
+	t.Run("Handshake - error advertisablea address", func(t *testing.T) {
+		var buffer1 bytes.Buffer
+		var buffer2 bytes.Buffer
+		stream1 := mock.NewStream(&buffer1, &buffer2)
+		stream2 := mock.NewStream(&buffer2, &buffer1)
+
+		testError := errors.New("test error")
+		aaddresser.err = testError
+		defer func() {
+			aaddresser.err = nil
+		}()
+
+		w, _ := protobuf.NewWriterAndReader(stream2)
+		if err := w.WriteMsg(&pb.SynAck{
+			Syn: &pb.Syn{
+				ObservedUnderlay: node1maBinary,
+			},
+			Ack: &pb.Ack{
+				Address: &pb.BzzAddress{
+					Underlay:  node2maBinary,
+					Overlay:   node2BzzAddress.Overlay.Bytes(),
+					Signature: node2BzzAddress.Signature,
+				},
+				NetworkID: networkID,
+				Light:     false,
+			},
+		}); err != nil {
+			t.Fatal(err)
+		}
+
+		res, err := handshakeService.Handshake(stream1, node2AddrInfo.Addrs[0], node2AddrInfo.ID)
+		if err != testError {
```

Comment:
use errors.Is?

Reply:
I don't expect error to  be wrapped here as test error is really what should be returned, so while being more nice, `error.Is` would be just a bit inaccurate. Do you think ti makes more sense to use it than exact compare